app/db/index.js

```diff
@@ -74,23 +79,30 @@ function addNewMarket(info) {
  * @param {*} info All question answers from the assessment page.
  */
 function updateExistingMarket(info) {
-  const marketsRef = db.child(info.marketInfo.marketName);
+  // Get current values to avoid overwriting current values.
+  getSpecificMarket(info.marketInfo.marketName).then((market) => {
+    const update = {
+      personalInfo: {
+        firstName: info.marketInfo.firstName,
+        lastName: info.marketInfo.lastName,
+        email: info.marketInfo.email,
+      },
+      marketInfo: {
+        marketName: market.marketInfo.marketName,
+        storeType: market.marketInfo.storeType,
+        address: market.marketInfo.address,
+        city: market.marketInfo.city,
+        state: market.marketInfo.state,
+        zip: market.marketInfo.zip,
+        marketLevel: parseInt(info.level, 10),
+      },
+      questions: info.questions,
+      missedQuestions: info.betterQuestions,
+    };
 
-  // Update market level.
-  marketsRef.child('marketInfo').update({
-    marketLevel: info.level,
+    // Send new and old values to update.
+    Market.updateOne({ _id: info.marketInfo.marketName }, { $set: update });
```

Comment:
Got it working, thanks for the tip there.  You'll see in the commits that there's an empty callback, but it's necessary otherwise it won't actually update.  Super weird but [this explains it](https://mongoosejs.com/docs/api.html#query_Query-findOneAndUpdate)